python/pyarrow/tests/test_pandas.py

```diff
@@ -255,6 +255,12 @@ def test_range_index_force_serialization(self):
         tm.assert_frame_equal(restored, df)
 
     def test_rangeindex_doesnt_warn(self):
+        if Version("2.2.0") <= Version(pd.__version__):
+            # make_block deprecation in pandas, still under discussion
+            # https://github.com/pandas-dev/pandas/pull/56422
+            # https://github.com/pandas-dev/pandas/issues/40226
+            pytest.skip("make_block deprecated in pandas 2.2.0")
```

Comment:
Alternatively, we could also specifically ignore this warning in the `with warnings.catch_warnings():` block below. (I don't know if it's easy to combine an "ignore" rule together with "error")

Reply:
This should work I think:

```
with warnings.catch_warnings():
    warnings.simplefilter("error")
    warnings.filterwarnings("ignore", "make_block is deprecated", DeprecationWarning)
    _check_pandas_roundtrip(df, preserve_index=True)
```